baseplate/lib/secrets.py

```diff
@@ -464,6 +467,88 @@ def _get_file_data(self, filename: str) -> Tuple[Dict, float]:
         return result
 
 
+class VaultCSIEntry(NamedTuple):
+    mtime: float
+    data: Any
+    # A mutex to prevent possible duplicate work from event loops
+    updating: bool
+
+
+class VaultCSISecretsStore(SecretsStore):
+    """Access to secret tokens with automatic refresh when changed.
+
+    This local vault allows access to the secrets cached on disk given a path to
+    a directory. It will automatically reload the cache when it is changed. Do not
+    cache or store the values returned by this class's methods but rather get
```

Comment:
:no_bell:  I would say, not caching or storing the values from secrets it's a good idea but also not a hard and fast rule. For example, if the secret never gets loaded from disk again due to the way a client system or library works, it may not be correct to say don't store the secret value. 

```suggestion
    a directory. It will automatically reload the cache when it was changed. Generally do not
    cache or store the values returned by this class's methods but rather get
```

Reply:
This was copypasta'd from the original implementation. I made this change as well as a few others